runtime/manta/src/migrations/assets_genesis.rs

```diff
@@ -0,0 +1,397 @@
+// Copyright 2020-2023 Manta Network.
+// This file is part of Manta.
+//
+// Manta is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Manta is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Manta.  If not, see <http://www.gnu.org/licenses/>.
+
+#![allow(clippy::unnecessary_cast)]
+
+use crate::sp_api_hidden_includes_construct_runtime::hidden_include::StorageHasher;
+use codec::Encode;
+use core::marker::PhantomData;
+use cumulus_primitives_core::ParaId;
+#[allow(deprecated)]
+use frame_support::migration::{
+    clear_storage_prefix, get_storage_value, put_storage_value, storage_key_iter,
+};
+use frame_support::{
+    dispatch::GetStorageVersion,
+    pallet_prelude::Weight,
+    traits::{Currency, Get, OnRuntimeUpgrade, StorageVersion},
+    Blake2_128Concat,
+};
+use manta_primitives::{
+    assets::{AssetConfig, AssetLocation, AssetRegistryMetadata},
+    types::{Balance, MantaAssetId},
+};
+use sp_runtime::BoundedVec;
+use sp_std::vec::Vec;
+
+pub type DepositBalanceOf<T, I = ()> = <<T as pallet_assets::Config<I>>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub const INITIAL_PALLET_ASSETS_MANAGER_VERSION: u16 = 0;
+pub const INITIAL_PALLET_ASSETS_VERSION: u16 = 0;
+
+pub struct AssetsGenesis<T>(PhantomData<T>);
+impl<T> OnRuntimeUpgrade for AssetsGenesis<T>
+where
+    T: pallet_asset_manager::Config + pallet_assets::Config,
+{
+    fn on_runtime_upgrade() -> Weight {
+        let asset_manager_storage_version =
+            <pallet_asset_manager::Pallet<T> as GetStorageVersion>::on_chain_storage_version();
+        let assets_storage_version =
+            <pallet_assets::Pallet<T> as GetStorageVersion>::on_chain_storage_version();
+        if asset_manager_storage_version != INITIAL_PALLET_ASSETS_MANAGER_VERSION
+            || assets_storage_version != INITIAL_PALLET_ASSETS_VERSION
+        {
+            log::info!("Aborting migration due to unexpected on-chain storage versions for pallet-assets-manager: {:?} and pallet-assets: {:?}. Expectation was: {:?} and {:?}.", asset_manager_storage_version, assets_storage_version, INITIAL_PALLET_ASSETS_MANAGER_VERSION, INITIAL_PALLET_ASSETS_VERSION );
+            return T::DbWeight::get().reads(2);
+        }
+
+        log::info!(target: "asset-manager", "Starting migration for AssetManager...");
+
+        let pallet_prefix: &[u8] = b"AssetManager";
+        let _ = clear_storage_prefix(pallet_prefix, b"AssetIdLocation", b"", None, None);
```

Comment:
Are you being explicit for documentation sake or does
`let _ = clear_storage_prefix(pallet_prefix, b"", b"", None, None);` not work?

Reply:
It doesn't unfortunately or i'm misusing it somehow :/

Reply:
Lemme try again though, before i resolve this comment.

Reply:
Nope :/